packages/flame/lib/src/effects/effects.dart

```diff
@@ -28,34 +35,91 @@ abstract class ComponentEffect<T extends BaseComponent> {
   bool isInfinite;
   bool isAlternating;
   final bool isRelative;
+  final bool removeOnFinish;
   final bool _initialIsInfinite;
   final bool _initialIsAlternating;
-  double? percentage;
+
+  /// The percentage of the effect that has passed including [preOffset] and
+  /// [postOffset].
+  double percentage = 0.0;
+
+  /// The outcome the curve function, only updates after [preOffset] and before
+  /// [postOffset]
   double curveProgress = 0.0;
+
+  /// How long time it takes for the effect to peak, which means right before it
+  /// starts any potential reversal or reset. Including offsets.
   double peakTime = 0.0;
+
+  /// The time passed since the start of the effect, it will start to decrease
+  /// after it has reached [peakTime] if [isAlternating] is true, and reset to
+  /// zero if it is not.
   double currentTime = 0.0;
+
+  /// When an effect reaches the end, and the beginning if it is alternating,
+  /// it will overshoot 0.0 and [peakTime], this time is added to the next time
+  /// step.
   double driftTime = 0.0;
+
+  /// Whether the effect is going forward or is reversing.
```

Comment:
Can we have an example of forward and backwards in the comment?

Reply:
Is it clearer now?

Reply:
Ahh right its clearer now thanks